lib/auto-languageclient.js

```diff
@@ -59,6 +59,10 @@ export default class AutoLanguageClient {
     throw Error('Must override startServerProcess to start language server process when extending AutoLanguageClient');
   }
 
+  newLinterPushV2Adapter(connection: ls.LanguageClientConnection): LinterPushV2Adapter {
```

Comment:
A general IoC/DI pattern sounds nice, but isn't it overkill for now? (I try to avoid speculative over-engineering.)